Automattic/jetpack
# Per plugin locks in Plugins//update endpoint

Update the plugins/%s/update endpoint to adhere to per plugin locks. There are several distinct functionalities that can trigger plugins to update. By implementing per plugin locks we ensure that if a plugin is already being updated in a separate request that we don't try to update the plugin in the current request.

Addresses https://wp.me/p9o2xV-1s0

#### Changes proposed in this Pull Request:

* Per plugin locks in the plugins/%s/update endpoint.

#### Does this pull request change what data or activity we track or use?


NO

#### Testing instructions:






No Unit Tests for this change as it is a silent skip of the plugin and does not trigger a failure response.

Manual Testing.
On your site install/activate an old version of a plugin.
Access shell and create a lock for the plugin. Where SLUG is the plugin's slug.
```
include_once ABSPATH . 'wp-admin/includes/class-wp-upgrader.php';
WP_Upgrader::create_lock( 'jetpack_SLUG' );
```
Make a request to the `/sites/%s/plugins/%s/update/` endpoint with the plugin to be updated.
Verify the plugin is in fact not updated.
Release the lock WP_Upgrader::release_lock( 'jetpack_SLUG' );
Make the same request to update.
Verify the plugin is in fact updated.

To make an update call please reference jpop-issue-6618. You will need to pass the plugins param so it is a bulk request.
